src/listener/utils/transforms/server-hardware.js

```diff
@@ -0,0 +1,72 @@
+/*
+(c) Copyright (2016-2017) Hewlett Packard Enterprise Development LP
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+import Resource from './resource';
+import { getDeviceName } from '../../../middleware/utils/lexer';
+
+export default class ServerHardware extends Resource {
+
+  constructor(oneViewResource) {
+    if (oneViewResource) {
+      super(oneViewResource);
+      this.name = oneViewResource.name;
+      this.formFactor = oneViewResource.formFactor;
+      this.powerState = oneViewResource.powerState;
+      this.serverProfileUri = oneViewResource.serverProfileUri;
+      this.serverProfileHyperlink = oneViewResource.serverProfileHyperlink;
+    }
+  }
+
+  buildSlackFields() {
+    let fields = [];
+    let hasProfile = false;
+    for (const field in this) {
+      if (field === 'name' || field === 'type' || field === 'status' || field.toLowerCase().includes('hyperlink') || field === 'serverProfileUri' || !this[field]) {
+        continue;
+      }
+      fields.push({
+        title: field,
+        short: true,
+        value: this[field]
+      });
+    }
+    if (this.serverProfileUri) {
+      fields.push({
+        title: 'Profile',
+        short: true,
+        value: '<' + this.serverProfileHyperlink + '|' + getDeviceName(this.serverProfileUri) + '>'
+      });
+      hasProfile = true;
+    }
+    if (!hasProfile) {
+      fields.push({
+        title: 'Profile',
+        short: true,
+        value: 'Available for deployment'
+      });
+    }
+    return fields;
+  }
+
+  //TODO
+  // buildHipChat() {}
```

Comment:
The HipChat implementation should remain the same.  Eventually some of the code from hipchat.js will need to be moved into the Resource classes.